### ISSUE TITLE
Webviews aren't being garbage collected when the containing page is popped

### ISSUE THREAD

**Issue by davidortinau** [xamarin/Xamarin.Forms#1430]
Migrated from https://bugzilla.xamarin.com/show_bug.cgi?id=55749

### Description
In the sample program, ThirdPage contains a webview defined in the xaml.  Nothing is done to the webview in the code.  Run the program and press the ThirdPage button, then the back arrow, 5 times and look at the memory allocations - there are 5 instances of the ThirdPage, WebView, WebViewRenderer, and WebClient (snapshot1 in the attached file).  Not tap the button to garbage collect.  The instances of ThirdPage and WebView drops to one, but WebViewRenderer and WebViewClient are still 5 (snapshot2).  Why aren't these garbage collected?

### Basic Information

- Version with issue: 2.3.4

### Reproduction Link

[MemoryTest0.zip](https://github.com/xamarin/Xamarin.Forms/files/1573721/MemoryTest0.zip)

**Comment by samhouts** [comment]
This issue doesn't seem to have had any activity in a long time. We're working on prioritizing issues and resolving them as quickly as we can. To help us get through the list, we would appreciate an update from you to let us know if this is still affecting you on the latest version of Xamarin.Forms, since it's possible that we may have resolved this as part of another related or duplicate issue. If we don't see any new activity on this issue in the next 30 days, we'll evaluate whether this issue should be closed. Thank you!

**Comment by czuck** [comment]
Yes, it does appear to still be an issue.  I've updated the test project to use the latest Forms and the memory usage in the profiler continues to go up, and doesn't drop when garbage collected.
sample project..
[MemoryTest0.zip](https://github.com/xamarin/Xamarin.Forms/files/4692310/MemoryTest0.zip)
here is the results from the profiler ..
[MemoryTest.zip](https://github.com/xamarin/Xamarin.Forms/files/4692315/MemoryTest.zip)

**Comment by samhouts** [comment]
Thank you, @czuck !!

**Comment by Xxshark888xX** [comment]
Hi, I'm facing the same issue, any update regarding this?

The WebView it's consuming ~350MB of RAM even after destroying the page.
